bin/p2-launch/main.go

```diff
@@ -110,12 +113,32 @@ func main() {
 		log.Fatalf("Could not install manifest %s: %s", manifest.ID(), err)
 	}
 
-	success, err := pod.Launch(manifest)
-	if err != nil {
-		log.Fatalf("Could not launch manifest %s: %s", manifest.ID(), err)
-	}
-	if !success {
-		log.Fatalln("Unsuccessful launch of one or more things in the manifest")
+	if *noStart {
+		// p2-one-shot puts launchables into OS images but not start them
+		// here we skip post activating and starting the pod, did all other stuffs like creating current symlink, and service builder files
+		launchables, err := pod.Launchables(manifest)
+		if err != nil {
+			log.Fatalf("Could not fetch launchables from the menifest %s: %s", manifest.ID(), err)
+		}
+		for _, launchable := range launchables {
+			err := launchable.MakeCurrent()
+			if err != nil {
+				log.Fatalf("Failed to create current symlink for pod %s: %s", manifest.ID(), err)
+			}
+		}
+		err = pod.BuildRunitServices(launchables, manifest)
```

Comment:
Did this end up being duplicated code from elsewhere or did we have to write this fresh in the end?

Reply:
Retracting this statement as I read the code.